packages/integrations/solid/src/dependencies.ts

```diff
@@ -8,7 +8,7 @@ export async function getSolidPkgsConfig(isBuild: boolean, astroConfig: AstroCon
 		isBuild,
 		viteUserConfig: astroConfig.vite,
 		isFrameworkPkgByJson(pkgJson) {
-			return containsSolidField(pkgJson.exports || {});
+			return containsSolidField(pkgJson.exports || {}) || !!(pkgJson.peerDependencies || {})['solid-js'];
```

Comment:
The logic here is a tiny bit hard to read with the numerous pipe. Reviewing this, I was actually confused that the code added was in the parameter of `containsSolidField` and it took me a few reads before getting the logic and that it wasn't a parameter. Maybe a comment could help?

Reply:
Yeah, that's very true. I'm actually going to hold off on this one until I have a chance to speak with @bluwy because maybe making all packages with a peerDep isn't the right call here. Might just want to hardcode this one instead.

Reply:
We'd probably want to align with `vite-plugin-solid`'s [implementation](https://github.com/solidjs/vite-plugin-solid/blob/66955a81de1e6b84e02271e4f8603500b45656cb/src/index.ts#L301-L308) too, so I think we should avoid this as the package would also have issues in other Vite SSR Solid setups.

Maybe a solution is to remove the `solid-js` string here:

https://github.com/withastro/astro/blob/dc062f6695ce577dc569781fc0678c903012c336/packages/integrations/solid/src/index.ts#L50

Or only apply it in dev.

Reply:
Sounds reasonable since that's what vite-plugin-solid does. I've update the PR to remove `'solid'` as a noExternal, seems to work from my testing.